app/screens/home/channel_list/servers/servers_list/index.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {useIntl} from 'react-intl';
+import {ListRenderItemInfo, StyleSheet, View} from 'react-native';
+import {FlatList} from 'react-native-gesture-handler';
+
+import {useServerUrl} from '@context/server';
+import {useIsTablet} from '@hooks/device';
+import BottomSheetContent from '@screens/bottom_sheet/content';
+
+import ServerItem from './server_item';
+
+import type ServersModel from '@typings/database/models/app/servers';
+
+type Props = {
+    servers: ServersModel[];
+}
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+    },
+    contentContainer: {
+        marginVertical: 4,
+    },
+});
+
+const keyExtractor = (item: ServersModel) => item.url;
+
+const ServerList = ({servers}: Props) => {
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const serverUrl = useServerUrl();
+
+    const onAddServer = useCallback(() => {
+        // eslint-disable-next-line no-console
+        console.log('Lets add a server');
+    }, [servers]);
+
+    const renderServer = useCallback(({item: t}: ListRenderItemInfo<ServersModel>) => {
+        return (
+            <ServerItem
+                isActive={t.url === serverUrl}
+                server={t}
+            />
+        );
+    }, [servers]);
```

Comment:
This does not depend on the servers (at least directly). It only depends on the `serverUrl`, which we consider in most cases immutable during the lifetime of the component (not 100% sure in this context).

Any re-rendering caused by changed in `servers` should be done already by the FlatList.

Reply:
well, yes.. this would only be true if a new server is added/removed which is not really going to happen